internal/controller/costmanagementmetricsconfig_controller_test.go

```diff
@@ -1448,4 +1453,125 @@ var _ = Describe("MetricsConfigController - CRD Handling", Ordered, func() {
+			}, timeout, interval).Should(ContainSubstring("Reports are being stored"))
+
+			Expect(fetched.Status.APIURL).To(Equal(defaultAPIURL))
+			Expect(fetched.Status.Upload.LastSuccessfulUploadTime.IsZero()).To(BeTrue())
+		})
+
+		It("should store reports when source validation fails", func() {
+			Expect(setup()).Should(Succeed())
+			instCopy.Spec.Source.SourceName = sourceName
+			instCopy.Spec.Source.CreateSource = &trueValue
+			createObject(ctx, instCopy)
+
+			fetched := &metricscfgv1beta1.MetricsConfig{}
+
+			Eventually(func() bool {
+				_ = k8sClient.Get(ctx, types.NamespacedName{Name: instCopy.Name, Namespace: namespace}, fetched)
+				return fetched.Status.Authentication.AuthenticationCredentialsFound != nil
+			}, timeout, interval).Should(BeTrue())
+
+			// Should get source validation error and block upload
+			Eventually(func() string {
+				_ = k8sClient.Get(ctx, types.NamespacedName{Name: instCopy.Name, Namespace: namespace}, fetched)
+				return fetched.Status.Upload.UploadError
+			}, timeout, interval).Should(ContainSubstring("Reports are being stored"))
+
+			Expect(fetched.Status.APIURL).To(Equal(defaultAPIURL))
+			Expect(fetched.Status.Source.SourceError).ToNot(BeNil())
+			Expect(fetched.Status.Upload.LastSuccessfulUploadTime.IsZero()).To(BeTrue())
+		})
+
+		It("should set generic upload error message when SourceError is empty", func() {
+			testInstance := instCopy.DeepCopy()
+			testInstance.Name = "test-generic-message"
+			testInstance.Spec.Source.SourceName = ""
+			testInstance.Spec.Upload.UploadToggle = &trueValue
+			testInstance.Spec.APIURL = validTS.URL
+
+			Expect(k8sClient.Create(ctx, testInstance)).Should(Succeed())
+			defer func() { _ = k8sClient.Delete(ctx, testInstance) }()
+
+			uploadDir := "tmp/koku-metrics-operator-reports/upload"
+			Expect(os.MkdirAll(uploadDir, 0755)).To(Succeed())
+			_, err := Copy(0644, "test_files/testFile.tar.gz", filepath.Join(uploadDir, "test-file.tar.gz"))
+			Expect(err).To(BeNil())
+			defer os.Remove(filepath.Join(uploadDir, "test-file.tar.gz"))
+
+			fetched := &metricscfgv1beta1.MetricsConfig{}
+			Eventually(func() bool {
+				_ = k8sClient.Get(ctx, types.NamespacedName{Name: testInstance.Name, Namespace: namespace}, fetched)
+				return fetched.Status.Source.SourceDefined != nil && !*fetched.Status.Source.SourceDefined
+			}, timeout, interval).Should(BeTrue())
+
+			fetched.Status.Source.SourceError = ""
+			fetched.Status.Packaging.PackagedFiles = []string{"test-file.tar.gz"}
+			Expect(k8sClient.Status().Update(ctx, fetched)).Should(Succeed())
+
+			Eventually(func() string {
+				_ = k8sClient.Get(ctx, types.NamespacedName{Name: testInstance.Name, Namespace: namespace}, fetched)
+				return fetched.Status.Upload.UploadError
+			}, timeout, interval).Should(Equal("Reports are being stored until a valid integration is configured in console.redhat.com"))
+		})
+
+		It("should return error when uploadFiles fails", func() {
```

Comment:
I'm not really sure this test is necessary. We already have upload specific tests elsewhere.